lib/Fhp/Response/Response.php

```diff
@@ -35,8 +41,8 @@ public function __construct($rawResponse)
         }
 
         $this->rawResponse = $rawResponse;
-        $this->response    = $this->unwrapEncryptedMsg($rawResponse);
-        $this->segments = explode("'", $rawResponse);
+        $this->response = $this->unwrapEncryptedMsg($rawResponse);
+        $this->segments = preg_split("#(')(?=[A-Z]{4,}:\d|')#", $rawResponse);
```

Comment:
I think it would make more sense to use
`#'(?=[A-Z]{4,}:\d)|'#` ?
The you would also remove the last "'" from the very last segment.